ivy/functional/frontends/torch/reduction_ops.py

```diff
@@ -251,3 +251,22 @@ def unique(input, sorted=True, return_inverse=False, return_counts=False, dim=No
         values.append(results.counts)
 
     return Results(*values)
+
+
+@to_ivy_arrays_and_back
+def mode(input, dim=-1, keepdim=False, *, out=None):
+    freq = {}

```

Comment:
You are not using `dim` and `keepdim` anywhere currently. I don't think the tests can be passing like this.